File: app/src/main/java/com/hypnoweb/hypnowebapp/di/modules/NetworkModule.java
package com.hypnoweb.hypnowebapp.di.modules;

import com.hypnoweb.hypnowebapp.catalog.repo.network.CatalogApi;
import com.hypnoweb.hypnowebapp.networking.NetworkFactory;

import dagger.Module;
import dagger.Provides;

/**
 * Created by stefan on 12/3/2017.
 */

@Module
public class NetworkModule {

    public NetworkModule() {
    }


    @Provides
    public CatalogApi getCatalogApi(NetworkFactory networkFactory) {
        return networkFactory.getCatalogApi();
    }
}

File: app/src/main/java/com/hypnoweb/hypnowebapp/di/modules/NetworkModule.kt
package com.hypnoweb.hypnowebapp.di.modules

import com.hypnoweb.hypnowebapp.catalog.repo.network.CatalogApi
import com.hypnoweb.hypnowebapp.networking.NetworkFactory
import dagger.Module
import dagger.Provides


@Module
class NetworkModule {

    @Provides
    fun getCatalogApi(networkFactory: NetworkFactory): CatalogApi {
        return networkFactory.catalogApi
    }
}
